astropy/_erfa/core.py.templ

```diff
@@ -118,46 +115,47 @@ def check_errwarn(statcodes, func_name):
         warnings.warn('ERFA function "' + func_name + '" yielded ' + wmsg, ErfaWarning)
 
 
-# <-------------------------trailing shape verification----------------------->
+# <------------------------structured dtype conversion------------------------>
 
-def check_trailing_shape(arr, shape, name):
-    try:
-        if arr.shape[-len(shape):] != shape:
-            raise Exception()
-    except:
-        raise ValueError("{0} must be of trailing dimensions {1}".format(name, shape))
+class D3Fix(numpy.ndarray):
+    # We do this to help ensure Quantity happily lets this pass...
```

Comment:
Could you provide more details in the comment here? I don't quite understand why this is needed.

Reply:
Yes, how could one... I added a very detailed comment.